### ISSUE TITLE
Constant iRule Updates from Controller

### ISSUE THREAD

**Issue by DiogoAndre** [F5Networks/k8s-bigip-ctlr#410]
#### Description
Hi,  I'm running the controller in our Kubernetes cluster with nothing currently deployed using it. I keep seeing the message bellow in the logs. Is this expected?

```
[2017-10-30 11:21:49,316 f5_cccl.resource.resource INFO] Updating ApiIRule: /Kubernetes/http_redirect_irule
[2017-10-30 11:22:19,304 f5_cccl.resource.resource INFO] Updating ApiIRule: /Kubernetes/http_redirect_irule
[2017-10-30 11:22:50,814 f5_cccl.resource.resource INFO] Updating ApiIRule: /Kubernetes/http_redirect_irule
[2017-10-30 11:23:19,300 f5_cccl.resource.resource INFO] Updating ApiIRule: /Kubernetes/http_redirect_irule
```

Thanks!
#### Kubernetes Version
v1.6.7

#### Controller Version
1.2.0

#### BIG-IP Version

BIG-IP 12.1.2 Build 1.0.271 Hotfix HF1

#### Diagnostic Information
__kubectl get deployment f5-k8s-controller --namespace kube-system -o yaml__
```apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "2"
  creationTimestamp: 2017-07-31T11:34:00Z
  generation: 2
  labels:
    app: k8s-bigip-ctlr
  name: f5-k8s-controller
  namespace: kube-system
  resourceVersion: "35623038"
  selfLink: /apis/extensions/v1beta1/namespaces/kube-system/deployments/f5-k8s-controller
  uid: 25fe4767-75e4-11e7-a734-005056be6631
spec:
  replicas: 1
  selector:
    matchLabels:
      app: k8s-bigip-ctlr
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: k8s-bigip-ctlr
      name: f5-k8s-controller
    spec:
      containers:
      - args:
        - --running-in-cluster=true
        - --bigip-url=$(BIGIP_URL)
        - --bigip-partition=$(BIGIP_PARTITION)
        - --bigip-username=$(BIGIP_USERNAME)
        - --bigip-password=$(BIGIP_PASSWORD)
        command:
        - /app/bin/k8s-bigip-ctlr
        env:
        - name: BIGIP_PARTITION
          valueFrom:
            secretKeyRef:
              key: partition
              name: bigip-credentials
        - name: BIGIP_USERNAME
          valueFrom:
            secretKeyRef:
              key: username
              name: bigip-credentials
        - name: BIGIP_PASSWORD
          valueFrom:
            secretKeyRef:
              key: password
              name: bigip-credentials
        - name: BIGIP_URL
          valueFrom:
            secretKeyRef:
              key: url_mn
              name: bigip-credentials
        image: f5networks/k8s-bigip-ctlr:1.2.0
        imagePullPolicy: IfNotPresent
        name: k8s-bigip-ctlr
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: 2017-07-31T11:34:00Z
    lastUpdateTime: 2017-07-31T11:34:00Z
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  observedGeneration: 2
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
```

__kubectl describe pod f5-k8s-controller-1173703806-4t3w6 --namespace kube-system__
```
Name:		f5-k8s-controller-1173703806-4t3w6
Namespace:	kube-system
Node:		mdvmsrv-k8swr-wrk7.mn.up.st/10.99.243.17
Start Time:	Mon, 30 Oct 2017 09:02:51 -0200
Labels:		app=k8s-bigip-ctlr
		pod-template-hash=1173703806
Annotations:	kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"kube-system","name":"f5-k8s-controller-1173703806","uid":"27e616e5-b017-11e7-af88...
Status:		Running
IP:		10.20.2.131
Controllers:	ReplicaSet/f5-k8s-controller-1173703806
Containers:
  k8s-bigip-ctlr:
    Container ID:	docker://97c611b0af371d3fe94d33ec7a00953e406a0f805ecc85613c79d32170666f21
    Image:		f5networks/k8s-bigip-ctlr:1.2.0
    Image ID:		docker-pullable://f5networks/k8s-bigip-ctlr@sha256:9e02034acc6c5165516786dbd81718161db4d721134f283454af6f7db2337035
    Port:
    Command:
      /app/bin/k8s-bigip-ctlr
    Args:
      --running-in-cluster=true
      --bigip-url=$(BIGIP_URL)
      --bigip-partition=$(BIGIP_PARTITION)
      --bigip-username=$(BIGIP_USERNAME)
      --bigip-password=$(BIGIP_PASSWORD)
    State:		Running
      Started:		Mon, 30 Oct 2017 09:03:46 -0200
    Ready:		True
    Restart Count:	0
    Environment:
      BIGIP_PARTITION:	<set to the key 'partition' in secret 'bigip-credentials'>	Optional: false
      BIGIP_USERNAME:	<set to the key 'username' in secret 'bigip-credentials'>	Optional: false
      BIGIP_PASSWORD:	<set to the key 'password' in secret 'bigip-credentials'>	Optional: false
      BIGIP_URL:	<set to the key 'url_mn' in secret 'bigip-credentials'>		Optional: false
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-m08ph (ro)
Conditions:
  Type		Status
  Initialized 	True
  Ready 	True
  PodScheduled 	True
Volumes:
  default-token-m08ph:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-m08ph
    Optional:	false
QoS Class:	BestEffort
Node-Selectors:	<none>
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From					SubObjectPath			Type		Reason		Message
  ---------	--------	-----	----					-------------			--------	------		-------
  17m		17m		1	default-scheduler							Normal		Scheduled	Successfully assigned f5-k8s-controller-1173703806-4t3w6 to mdvmsrv-k8swr-wrk7.mn.up.st
  17m		17m		1	kubelet, mdvmsrv-k8swr-wrk7.mn.up.st	spec.containers{k8s-bigip-ctlr}	Normal		Pulling		pulling image "f5networks/k8s-bigip-ctlr:1.2.0"
  16m		16m		1	kubelet, mdvmsrv-k8swr-wrk7.mn.up.st	spec.containers{k8s-bigip-ctlr}	Normal		Pulled		Successfully pulled image "f5networks/k8s-bigip-ctlr:1.2.0"
  16m		16m		1	kubelet, mdvmsrv-k8swr-wrk7.mn.up.st	spec.containers{k8s-bigip-ctlr}	Normal		Created		Created container with id 97c611b0af371d3fe94d33ec7a00953e406a0f805ecc85613c79d32170666f21
  16m		16m		1	kubelet, mdvmsrv-k8swr-wrk7.mn.up.st	spec.containers{k8s-bigip-ctlr}	Normal		Started		Started container with id 97c611b0af371d3fe94d33ec7a00953e406a0f805ecc85613c79d32170666f21
```

**Comment by DiogoAndre** [comment]
I also see this in the BIG-IP Logs:

```
pid=11795 user=***** folder=/Kubernetes module=(tmos)# status=[Command OK] cmd_data=modify ltm rule /Kubernetes/http_redirect_irule { "when HTTP_REQUEST { HTTP::redirect https://[getfield [HTTP::host] \":\" 1]:443[HTTP::uri] } " app-service none }:
```

**Comment by f5gary** [comment]
The root cause for this issue is described in https://github.com/f5devcentral/f5-cccl/issues/183. We will likely fix it there as well as in this repo where the IRules are defined so they do not contain leading whitespace.

**Comment by DiogoAndre** [comment]
Thanks for the quick diagnostic folks!

**Comment by DiogoAndre** [comment]
Hey.. do you have an estimate for when a new version with this fix will be released? Thanks!

**Comment by f5gary** [comment]
@DiogoAndre the fix is in the code now, and it will be included in our next released version, 1.4. This release should occur some time around Dec 31.
